### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by Gh-Reza**
### In this capstone:

- Created the project using create-react-app
- Installed the dependencies
- Used three APIs
     - [Weather API](https://www.weatherapi.com/)
     - [Rest Countries API](https://restcountries.com/)
     - [Countries Now API](https://countriesnow.space/)
- Added the pages with their components:
     - Header: Search
     - Home page: country
     - Details page: State
- Styled the components using Bootstrap and CSS
- Added the tests for each component and the Redux functions
- Updated the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,167 @@
...
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+Click [here](https://weather-app-ln0n.onrender.com/)
+
+## Presentation
+
+[Watch the video](https://drive.google.com/file/d/1XDVBzCsGMfe4i34RJmy5X1Z7LC6WFHNb/view?usp=sharing)
```
Comment (Lucas-Masaba) [inline README.md]: - [ ] Please mention a piece of code you find interesting or you are proud of in your presentation video as mentioned in the project requirements. You only showed the demo and mentioned the features of the project.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import Home from './pages/Home';
+import Details from './pages/Details';
+
+function App() {
+  return (
+    <div className="App d-flex flex-column min-vh-100">
+      <Header />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/details" element={<Details />} />
```
Comment (Lucas-Masaba) [inline src/App.js]: - [ ] Please make sure each detail page a user opens should have a unique route. Currently, when I click on multiple countries, the routes are all the same leading to `/details` as seen in the screenshot below
image

This route should look like this and I suggest that your parameter should be `id`
`<Route exact path="/item/:id" component={Item} />`

File: src/components/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { NavLink, useLocation } from 'react-router-dom';
+import Icon from '../assets/icon.png';
+import BackIcon from '../assets/icon50.png';
+import Search from './Search';
+
+const Header = () => {
+  const location = useLocation();
+  const isDetailsPage = location.pathname === '/details';
+  return (
+    <header className="header text-white d-flex justify-content-between align-items-center">
```
Comment (Lucas-Masaba) [inline src/components/Header.jsx:9]: - [ ] According to the design guidelines, you should follow the 
  * Typography: font face, size, and weight
  * Layout: composition and space between elements.
Please add the nav bar as seen in the original design. It's layout should be similar.
### Your design 
image
### Original
image

File: src/pages/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    dispatch(getCountryStates(country));
+  };
+
+  useEffect(() => {
+    dispatch(getCountries());
+  }, [dispatch]);
+
+  const countries = useSelector((store) => store.countries.countries);
+  return (
+    <main className="gen container container-md-fluid">
```
Comment (Lucas-Masaba) [inline src/pages/Home.jsx]: - [ ] Place place these cards together as seen in the original design. Currently, you have left spaces between each card
### Your Implementation
image
### Required
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Gh-Reza  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!
almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Professional README :scroll: 
- Good-looking and responsive UI :star2: 
- All tests passing :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Route, Routes } from 'react-router-dom';
+import Header from './components/Header';
+import './App.css';
```
Comment (tufoinnkuo10) [inline src/App.js:3]: - [optional] please remember this is a mobile version project. Ensure it's responsive till 768px where it switches. From 580px it switches already.
- [optional] kindly add the small arrows on the cards, so it indicates that it should be clicked and as well follow the original design
- [optional] please increase the size of the search bar for user-friendly purposes.
Screenshot (1287)
Screenshot (1288)

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Gh-Reza 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Design well followed
- Test ok
- work well presented
- No linter errors


Video presentation
----  Please always introduce your self and give a clear outline of what you did, why you built it and how important it will be of help. add your volume and check before recording. nice work

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**